### PR TITLE
Extend attributes to Rows and Columns

### PR DESCRIPTION
**Pull request by zackpyle**
We can use fl_builder_column_attributes and fl_builder_row_attributes to extend this functionality to Beaver Builder rows and columns.

### CODE REVIEW

**Comment by zackpyle** [comment]
Updated to latest version

File: bb-custom-attributes.php [hunk 1]
```diff
@@ -21,8 +21,10 @@ class BBCustomAttributes
     public function load()
     {
         add_action('plugins_loaded', [$this, 'registerForm']);
-        add_filter('fl_builder_register_settings_form', [$this, 'filterAdvancedModule'], 10, 2);
-        add_filter('fl_builder_module_attributes', [$this, 'filterAttributes'], 10, 2);
+        add_filter('fl_builder_register_settings_form', [$this, 'filterAdvancedTabAttr'], 10, 2);
+        add_filter('fl_builder_module_attributes', [$this, 'filterModuleAttributes'], 10, 2);
+        add_filter('fl_builder_column_attributes', [$this, 'filterColAttributes'], 10, 2);
+		add_filter('fl_builder_row_attributes', [$this, 'filterRowAttributes'], 10, 2);
```
Comment (JasonTheAdams) [inline bb-custom-attributes.php:27]: Some formatting funkiness here. It's using tabs instead of spaces.

File: bb-custom-attributes.php [hunk 2]
```diff
@@ -93,6 +95,28 @@ public function filterAdvancedModule($form, $id)
...
+                'type'         => 'form',
+                'form'         => 'custom_attributes',
+                'label'        => __('Attributes'),
+                'help'         => __('Adds custom attributes to the column'),
+                'multiple'     => true,
+                'preview_text' => 'key'
+            ];
+        }
+		
+		if('row' === $id ) {
```
Comment (JasonTheAdams) [inline bb-custom-attributes.php:110]: Another instance of tabs

File: bb-custom-attributes.php [hunk 3]
```diff
@@ -105,13 +129,50 @@ public function filterAdvancedModule($form, $id)
...
+    public function filterRowAttributes($attributes, $row)
+    {
+        if (isset($row->settings->custom_attributes)) {
+            foreach ($row->settings->custom_attributes as $attribute) {
+                $key = esc_attr($attribute->key);
+                if ('yes' === $attribute->override || ! isset($attributes[$key])) {
+                    $value = do_shortcode(esc_attr($attribute->value));
+                    $attributes[$key] = $value;
                 }
             }
```
Comment (JasonTheAdams) [inline bb-custom-attributes.php:177]: Do these need to be three separate filter functions? Unless I'm missing something the only difference is the name of the second parameter.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left a few comments for changes.

File: bb-custom-attributes.php (hunk 3 above)
Comment (zackpyle) [reply bb-custom-attributes.php:177]: You are correct. I didn't even notice they were the same! I'll merge them

**Comment by zackpyle** [comment]
Updated. Let me know your thoughts!

File: bb-custom-attributes.php [hunk 4]
```diff
@@ -81,37 +83,60 @@ public function registerForm()
...
     {
-        if (isset($module->settings->custom_attributes)) {
-            foreach ($module->settings->custom_attributes as $attribute) {
-                $key = esc_attr($attribute->key);
-                if ('yes' === $attribute->override || ! isset($attributes[$key])) {
-                    $attributes[$key] = esc_attr($attribute->value);
+        if (isset($element->settings->custom_attributes)) {
+		    foreach ($element->settings->custom_attributes as $attribute) {
+		        $key = esc_attr($attribute->key);
+		        if ('yes' === $attribute->override || !isset($attributes[$key])) {
```
Comment (JasonTheAdams) [inline bb-custom-attributes.php:137]: Mix of tabs and spaces here

**Review by JasonTheAdams — APPROVED** [review]
Much better @zackpyle! I just left one minor comment. Please fix that formatting and then we'll merge this!

**Comment by zackpyle** [comment]
@JasonTheAdams Sorry about that. I'm a tabs guy haha. All fixed

**Review by zackpyle — COMMENTED** [review]
@JasonTheAdams Sorry about that. I'm a tabs guy haha. All fixed

**Comment by JasonTheAdams** [comment]
Thanks so much for your contribution, @zackpyle!

**Pull request merged**